File: app/src/main/java/com/karthik/splash/Adapters/LibraryAdapter.java
package com.karthik.splash.Adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.karthik.splash.Models.Library.Library;
import com.karthik.splash.R;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by karthikrk on 03/12/17.
 */

public class LibraryAdapter extends RecyclerView.Adapter<LibraryAdapter.LibraryHolder>{
    private ArrayList<Library> libraries;

    public LibraryAdapter(ArrayList<Library> libraries){
        this.libraries = libraries;
    }

    @Override
    public LibraryAdapter.LibraryHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View row = LayoutInflater.from(parent.getContext()).inflate(R.layout.library_row,
                parent,false);
        return new LibraryHolder(row);
    }

    @Override
    public void onBindViewHolder(LibraryAdapter.LibraryHolder holder, int position) {
        Context context = holder.libTitle.getContext();
        Library library = libraries.get(position);
        holder.libTitle.setText(library.libName);
        holder.libVersion.setText(context.getString(R.string.Version)+" "+library.version);
    }

    @Override
    public int getItemCount() {
        return libraries.size();
    }

    class LibraryHolder extends RecyclerView.ViewHolder{
        @BindView(R.id.libtitle)
        TextView libTitle;
        @BindView(R.id.libversion)
        TextView libVersion;
        public LibraryHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/com/karthik/splash/aboutscreen/LibraryAdapter.kt
package com.karthik.splash.aboutscreen


import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.karthik.splash.Models.Library.Library
import com.karthik.splash.R

import java.util.ArrayList

/**
 * Created by karthikrk on 03/12/17.
 */

class LibraryAdapter(private val libraries: ArrayList<Library>) : RecyclerView.Adapter<LibraryAdapter.LibraryHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LibraryHolder {
        val row = LayoutInflater.from(parent.context).inflate(R.layout.library_row,
                parent, false)
        return LibraryHolder(row)
    }

    override fun onBindViewHolder(holder:LibraryHolder,position: Int) {
        val context = holder.libtitle.context
        val library = libraries[position]
        holder.libtitle.text = library.libName
        holder.libversion.text = "${context.getString(R.string.Version)} ${library.version}"
    }

    override fun getItemCount() = libraries.size

    inner class LibraryHolder:RecyclerView.ViewHolder {
        val libtitle:TextView
        val libversion:TextView
        constructor(itemView: View):super(itemView){
            libtitle = itemView.findViewById(R.id.libtitle)
            libversion = itemView.findViewById(R.id.libversion)
        }
    }
}
